java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -40,17 +41,29 @@
 
 import javax.annotation.Nullable;
 
+import java.util.List;
+
 @Rule(key = "S3546")
 @RuleTemplate
 public class CustomUnclosedResourcesCheck extends SECheck {
 
-  static class ResourceStatus implements ObjectConstraint.Status {
+  //see SONARJAVA-1624 class cannot be static, different classes are needed for every instance of this template rule
+  public class CustomResourceConstraint implements Constraint {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)